### DISCUSSION TITLE
Getting a list of schema fields with validation rules

### DISCUSSION THREAD

**Discussion by yudinmaxim** [open-circle/valibot#391] [Q&A]
I couldn't find a way to programmatically determine which fields are required and which can be ignored based on the schema. For example, if using yup, it is done in the following way: 
```javascript
// If I use yup
// Getting the schema description
const schemaDescription = dataSchema.describe();

// A function to determine the required fields based on the description
const getRequiredFields = (description) => { 
  const requiredFields = {};
  Object.entries(description.fields).forEach(([key, value]) => {
    // Check if the field contains the 'required' key 
    requiredFields[key] = value.tests.some(test => test.name === 'required');
  });
  return requiredFields;
};

```
How can the same be achieved using valibot?

**Comment by fabian-hiller** [comment]
What is your use case? Fields are required by default in Valibot, but you can make them optional with `optional`. With `Schema.entries` you can access the entries of an object schema.

**Reply by yudinmaxim** [reply]
I'm using Valibot along with vee-validate for form validation in my Vue application. I want to use the required attribute in the form template to mark fields that are mandatory for submission.

**Reply by fabian-hiller** [reply]
You could try the following:

```ts
const getRequiredFields = (description) => { 
  const requiredFields = {};
  Object.entries(ObjectSchema.entries).forEach(([key, schema]) => {
    requiredFields[key] = 'type' in schema && schema.type !== 'optional';
  });
  return requiredFields;
};
```

**Reply by yudinmaxim** [reply]
Thank you for the answer.
Yes - it really makes it possible to define the fields that are required.
Thanks a lot

**Comment by ruiaraujo012** [comment]
I'm looking for that exact thing but, I'm using unions and variants, ho can I do it? Any idea?

Thanks.

**Reply by fabian-hiller** [reply]
Can you provide more details and a code example?

**Reply by ruiaraujo012** [reply]
I opened a new discussion here https://github.com/fabian-hiller/valibot/discussions/484, maybe we can continue there?
